src/main/java/searching_system/SearcherManager.java

```diff
@@ -0,0 +1,35 @@
+package searching_system;
+
+import data_access_storage.*;
+import entities.*;
+import matching_system.*;
+
+import java.util.ArrayList;
+
+public class SearcherManager implements SearcherInputBoundary{
+    private final Searcher searcher;
+
+    final UserRepoManager userRepoManager;
+
+    final SearcherOutputBoundary searcherOutputBoundary;
+
+    public SearcherManager(SearcherOutputBoundary searcherOutputBoundary, SearchRequestModel searchRequestModel, UserRepoManager userRepoManager){
+        this.userRepoManager = userRepoManager;
+        this.searcherOutputBoundary = searcherOutputBoundary;
+        ArrayList<UserAccount> userAccountArrayList = new ArrayList<UserAccount>();
+        for(RequestModel rm: this.userRepoManager.getAllUserAccount().values()){
+            userAccountArrayList.add(rm.getUserAccount());
+        }
+        this.searcher = new Searcher(searchRequestModel.getKeyword(), userAccountArrayList);
+
+    }
+    @Override
+    public SearchResponseModel create(SearchRequestModel requestModel){
+
+        ArrayList<UserAccount> searchedUsers = searcher.getSearches();
+
+        return new SearchResponseModel(searchedUsers);
+    }
+
```

Comment:
This method takes the parameter requestModel, but does not use it?   Based on your code structure, 
this method is the method that does the real job: takes input from controllers and reflect the user cases that the user-at-the-keyword wants to do with your program. Suggest to revise this method together with the constructor in this class.